File: src/test/java/no/nav/sosialhjelp/soknad/consumer/unleash/ByInstanceIdStrategyTest.java
package no.nav.sosialhjelp.soknad.consumer.unleash;

import org.junit.jupiter.api.Test;

import java.util.Map;

import static org.assertj.core.api.Assertions.assertThat;

class ByInstanceIdStrategyTest {

    private ByInstanceIdStrategy strategy;

    @Test
    void shouldReturnFalse_instanceIdNotInMap() {
        strategy = new ByInstanceIdStrategy("local");
        var parameters = Map.of("instance.id", "dev-sbs,dev-sbs-intern");

        assertThat(strategy.isEnabled(parameters)).isFalse();
    }

    @Test
    void shoudReturnTrue_instanceIdInMap() {
        strategy = new ByInstanceIdStrategy("dev-sbs");
        var parameters = Map.of("instance.id", "dev-sbs,dev-sbs-intern");

        assertThat(strategy.isEnabled(parameters)).isTrue();
    }

}

File: src/test/kotlin/no/nav/sosialhjelp/soknad/client/featuretoggle/ByInstanceIdStrategyTest.kt
package no.nav.sosialhjelp.soknad.client.featuretoggle

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class ByInstanceIdStrategyTest {

    @Test
    fun shouldReturnFalse_instanceIdNotInMap() {
        val byInstanceIdStrategy = ByInstanceIdStrategy("local")
        val parameters = mutableMapOf("instance.id" to "dev-sbs,dev-sbs-intern")
        assertThat(byInstanceIdStrategy.isEnabled(parameters)).isFalse
    }

    @Test
    fun shoudReturnTrue_instanceIdInMap() {
        val byInstanceIdStrategy = ByInstanceIdStrategy("dev-sbs")
        val parameters = mutableMapOf("instance.id" to "dev-sbs,dev-sbs-intern")
        assertThat(byInstanceIdStrategy.isEnabled(parameters)).isTrue
    }
}
